src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,37 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const status = useAppSelector(state => state.filter.status);
+  const query = useAppSelector(state => state.filter.query);
+
+  const handleStatus = (value: Status) => {
+    dispatch(filterActions.status(value));
+  };
+
+  const handleQuery = (value: string) => {
+    if (!value.trim()) {
+      dispatch(filterActions.query(''));
+    } else {
+      dispatch(filterActions.query(value));
+    }
```

Comment:
```suggestion
    dispatch(filterActions.query(value));
```
Redundant code. What if I want to found a series of spaces? :)